tests/lru_cache_test.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import string
+import tempfile
+
+from absl.testing import absltest
+
+from jax._src.lru_cache import LRUCacheImpl
+import jax._src.test_util as jtu
+
+
+class LRUCacheTest(jtu.JaxTestCase):
+
+    def test_cache_eviction(self):
+        def generate_random_k():
+            # simulate keys with many collisions
+            return random.choice(string.ascii_lowercase[:12])
+
+        def generate_random_v(k):
+            # simulate large values while ensures that one k corresponds to one v
+            return bytes(f"{k}abcdefghijklmnopqrstuvwxyz" * 8192, encoding="utf-8")
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            cache = LRUCacheImpl(tmpdirname, 884700)
+
+            for _ in range(100000):
+                k = generate_random_k()
```

Comment:
I think it's fine not do a random test here. You can have a small cache (e.g. `max_size=2`) and 3-4 different keys.

Reply:
Yes, please don't use randomness here. Instead, manually put and get so you know exactly what's happening and can test that the cache is working as intended. Like make the cache size relatively small, and make sure you test cases like:

1. No entry is read, so the least recent put should be deleted
2. The least recent put is read, so a different put should be deleted
3. Nothing is deleted before the cache limit is met

Here's an old version of LRU eviction tests (that have since been deleted) for inspiration: https://github.com/google/jax/commit/4bae894862babdae09967a6503f4912eded1c961#diff-025b466784d0643aeb37e6042b587b2b15a5bc50d1a0fcf31be40d3e42955357

Reply:
Done